staging/src/k8s.io/client-go/rest/with_retry.go

```diff
@@ -147,34 +164,84 @@ func (r *withRetry) NextRetry(req *http.Request, resp *http.Response, err error,
 	//     need to check if it is retryable
 	seconds, wait := checkWait(resp)
 	if !wait {
-		return retryAfter, false
+		return false
 	}
 
-	retryAfter.Wait = time.Duration(seconds) * time.Second
-	retryAfter.Reason = getRetryReason(r.attempts, seconds, resp, err)
-	return retryAfter, true
+	r.retryAfter.Wait = time.Duration(seconds) * time.Second
+	r.retryAfter.Reason = getRetryReason(r.attempts, seconds, resp, err)
+	return true
 }
 
-func (r *withRetry) BeforeNextRetry(ctx context.Context, backoff BackoffManager, retryAfter *RetryAfter, url string, body io.Reader) error {
+func (r *withRetry) PrepareForNextRetry(ctx context.Context, request *Request) error {
```

Comment:
the pattern seems to be always
```
				if r.retry.IsNextRetry(req, resp, err, neverRetryError) {
					err := r.retry.PrepareForNextRetry(ctx, r)
					if err == nil {
						return false, nil
					
```

and this only does 3 additional checks, the new one is just check the output of `IsNextRetry` that is set r.retryAfter
Should we merge this 2 methods? is PrepareForNextRetry really needed now?

Reply:
that's a good suggestion, i combined these two, now `Stream`, `Watch`, and `Do` look more easy to follow